### PR TITLE
feat: update fuels deploy contracts to beta-5

### LINKED ISSUE #66: Add test for deposit erc20 from message input
bridgeERC20.ts is testing the deposit erc20 process, but when mounting the inputs to pay the transaction, it only considers Coin inputs.

- should add a test to validate if deposit erc20 token from CoinMessage works
  - bridge (deposit) ETH to a fuel account
  - then bridge(deposit) erc20 to the same fuel account
```
export function resourcesToInputs(resources: Array<Resource>) {
  const coinResources: Coin[] = resources.filter((r) =>
    isCoin(r)
  ) as unknown as Coin[];

  const coinInputs: Array<TransactionRequestInput> = coinResources.map(
    (r: Coin) => ({
      type: InputType.Coin,
      id: r.id,
      owner: r.owner.toB256(),
      amount: r.amount.toHex(),
      assetId: r.assetId,
      txPointer: ZeroBytes32,
      witnessIndex: 0,
    })
  );
  const messageCoinResources: MessageCoin[] = resources.filter((r) =>
    isMessage(r)
  ) as unknown as MessageCoin[];

  const messageCoinInputs: Array<TransactionRequestInput> =
    messageCoinResources.map((r: MessageCoin) => ({
      type: InputType.Message,
      assetId: r.assetId,
      sender: r.sender.toB256(),
      recipient: r.recipient.toB256(),
      nonce: r.nonce,
      amount: r.amount.toHex(),
      daHeight: r.daHeight.toHex(),
      witnessIndex: 0,
    }));
  // return coinInputs;
  return [...coinInputs, ...messageCoinInputs];
}

```

### PR DESCRIPTION
**Pull request by LuizAsFight**
### Objective
Update this repo to beta-5

- [ ] fuel-core `0.22.0`
- [ ] fuels-ts `next`

### Additional features
- [x] Use `forc` from TS SDK, instead of directly installing in the project
- [x] Add release-pr action, which publishes the packages to `npm` with `pr-106` version
- [x] Closes #66

### CODE REVIEW

File: Cargo.toml
Comment (Voxelot) [inline Cargo.toml:25]:
```suggestion
fuel-core-types = { version = "0.22.0", default-features = false }
```

File: Cargo.toml
Comment (Voxelot) [inline Cargo.toml:28]:
```suggestion
fuels = { version = "0.54.0", features = ["fuel-core-lib"] }
```

**Comment by Voxelot** [comment]
Why are we using `next` instead of versioned releases of the fuel-ts sdk that correspond to compatible network versions?

**Comment by DefiCake** [comment]
> Why are we using `next` instead of versioned releases of the fuel-ts sdk that correspond to compatible network versions?

I am guessing that we are waiting for a new ts-sdk release. @LuizAsFight can confirm.

**Review by DefiCake — DISMISSED** [review]

File: packages/fungible-token/bridge-fungible-token/src/cast.sw
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/src/cast.sw:16]: Should raise an issue, and write in a TODO, prior to merging so that this isn't forgotten

File: packages/fungible-token/bridge-fungible-token/src/data_structures.sw
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/src/data_structures.sw:42]: I would consider this as a fmt bug. Perhaps raising an issue for discussion is good

File: packages/fungible-token/bridge-fungible-token/src/main.sw
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/src/main.sw:1]: formatter! i hate you!

File: packages/fungible-token/bridge-fungible-token/src/main.sw
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/src/main.sw:334]:
```suggestion
    let previous_amount = storage.refund_amounts.get(from).get(asset).try_read().unwrap_or(ZERO_B256).as_u256();
```

Compiler shouldn't force the annotation here afaik

File: packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs:219]:
```suggestion
```

We ought to remove these from `master`

File: packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs:455]:
```suggestion
```

File: packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/tests/functions/bridge/mod.rs:461]:
```suggestion
```

Kinda unsure what is going on now. Lots of debugging

File: packages/fungible-token/bridge-fungible-token/tests/utils/setup.rs
Comment (Braqzen) [inline packages/fungible-token/bridge-fungible-token/tests/utils/setup.rs:294]: el?

What is that? Let's use `lol` instead? :)

File: packages/integration-tests/scripts/bridgeERC20.ts
Comment (Braqzen) [inline packages/integration-tests/scripts/bridgeERC20.ts:35]: What's the reason for changing the signer?

File: packages/integration-tests/scripts/bridgeERC20.ts
Comment (Braqzen) [inline packages/integration-tests/scripts/bridgeERC20.ts:178]: Has there been a renaming?
Seems like a `messageId` and `nonce` may be 2 different things.

File: packages/message-predicates/Forc.lock
Comment (Braqzen) [inline packages/message-predicates/Forc.lock:20]: Weren't we using 0.48.1 above? Why is this updated to 0.47.0?

File: packages/message-predicates/contract-message-predicate/tests/utils/builder.rs
Comment (Braqzen) [inline packages/message-predicates/contract-message-predicate/tests/utils/builder.rs:3]:
```suggestion
```

Can remove?

**Review by Braqzen — COMMENTED** [review]
lol, what are these changeset names :)

File: packages/integration-tests/scripts/bridgeERC20.ts
Comment (LuizAsFight) [reply packages/integration-tests/scripts/bridgeERC20.ts:178]: yes they are different. change was intentional

Comment (LuizAsFight) [reply packages/integration-tests/scripts/bridgeERC20.ts:178]: as sdk now expects nonce

File: docker/fuel-core/fuel_core.sh
Comment (LuizAsFight) [reply docker/fuel-core/fuel_core.sh:4]: I will decrease the increase :) it was increased because of slow connection at the offsite...

File: packages/integration-tests/scripts/bridgeERC20.ts
Comment (LuizAsFight) [reply packages/integration-tests/scripts/bridgeERC20.ts:35]: I'm not sure

cc @DefiCake

Comment (DefiCake) [reply packages/integration-tests/scripts/bridgeERC20.ts:35]: The block committer is using signer[1]. This is one of the couple of reasons the tests were showing flakiness, because if the wallet trying to transact with the bridges and the block committer send transactions close to each other, there will be nonce reusability issues

File: packages/message-predicates/Forc.lock
Comment (DefiCake) [reply packages/message-predicates/Forc.lock:20]: Not really sure about what happened with this one. Apparently forc does not update the deps automatically from the workspace. I 'll raise it just in case too

File: packages/fungible-token/bridge-fungible-token/src/data_structures.sw
Comment (JoshuaBatty) [reply packages/fungible-token/bridge-fungible-token/src/data_structures.sw:42]: Yeah this looks like a formatting bug. @Braqzen would you mind opening an issue for this?

File: packages/fungible-token/bridge-fungible-token/tests/utils/setup.rs
Comment (DefiCake) [reply packages/fungible-token/bridge-fungible-token/tests/utils/setup.rs]: Fixed

File: packages/fungible-token/bridge-fungible-token/src/cast.sw
Comment (DefiCake) [reply packages/fungible-token/bridge-fungible-token/src/cast.sw:16]: Seems to be gone in 0.48, most likely this was a 0.47 issue

File: packages/message-predicates/Forc.lock
Comment (DefiCake) [reply packages/message-predicates/Forc.lock]: Fixed

File: packages/fungible-token/bridge-fungible-token/src/data_structures.sw
Comment (Braqzen) [reply packages/fungible-token/bridge-fungible-token/src/data_structures.sw:42]: Sure, I'll create a minimal repro today

**Review by Braqzen — APPROVED** [review]
This beta(5) be good :)

**Pull request merged**
